unit_test/batched/sparse/Test_Batched_TeamCG.hpp

```diff
@@ -0,0 +1,206 @@
+/// \author Kim Liegeois (knliege@sandia.gov)
+
+#include "gtest/gtest.h"
+#include "Kokkos_Core.hpp"
+#include "Kokkos_Random.hpp"
+
+//#include "KokkosBatched_Vector.hpp"
+
+#include "KokkosBatched_CG.hpp"
+
+#include "KokkosKernels_TestUtils.hpp"
+
+#include "KokkosBatched_CG.hpp"
+
+#include "KokkosBatched_CrsMatrix.hpp"
+
+#include "Test_Batched_SparseUtils.hpp"
+
+using namespace KokkosBatched;
+
+namespace Test {
+namespace TeamCG {
+
+template <typename DeviceType, typename ValuesViewType, typename IntView,
+          typename VectorViewType>
+struct Functor_TestBatchedTeamCG {
+  const ValuesViewType _D;
+  const IntView _r;
+  const IntView _c;
+  const VectorViewType _X;
+  const VectorViewType _B;
+  const int _N_team;
+  KrylovHandle<typename ValuesViewType::value_type> *handle;
+
+  KOKKOS_INLINE_FUNCTION
+  Functor_TestBatchedTeamCG(const ValuesViewType &D, const IntView &r,
+                            const IntView &c, const VectorViewType &X,
+                            const VectorViewType &B, const int N_team)
+      : _D(D), _r(r), _c(c), _X(X), _B(B), _N_team(N_team) {
+    handle = new KrylovHandle<typename ValuesViewType::value_type>;
+  }
+
+  template <typename MemberType>
+  KOKKOS_INLINE_FUNCTION void operator()(const MemberType &member) const {
+    const int first_matrix = static_cast<int>(member.league_rank()) * _N_team;
+    const int N            = _D.extent(0);
+    const int last_matrix =
+        (static_cast<int>(member.league_rank() + 1) * _N_team < N
+             ? static_cast<int>(member.league_rank() + 1) * _N_team
+             : N);
+
+    auto d = Kokkos::subview(_D, Kokkos::make_pair(first_matrix, last_matrix),
+                             Kokkos::ALL);
+    auto x = Kokkos::subview(_X, Kokkos::make_pair(first_matrix, last_matrix),
+                             Kokkos::ALL);
+    auto b = Kokkos::subview(_B, Kokkos::make_pair(first_matrix, last_matrix),
+                             Kokkos::ALL);
+
+    using Operator = KokkosBatched::CrsMatrix<ValuesViewType, IntView>;
+
+    Operator A(d, _r, _c);
+
+    KokkosBatched::TeamCG<MemberType>::template invoke<Operator,
+                                                       VectorViewType>(
+        member, A, b, x, handle);
+  }
+
+  inline void run() {
+    typedef typename ValuesViewType::value_type value_type;
+    std::string name_region("KokkosBatched::Test::TeamCG");
+    std::string name_value_type =
```

Comment:
Also you can make the above `static inline constexpr` and it would become a compile time decision based on the template parameter, no longer a runtime one.